tests/ci.sh

```diff
@@ -111,62 +128,58 @@ RUST_BACKTRACE=1 cargo test --doc $FEATURES
 
 echo "Start Parsec for integration tests"
 RUST_LOG=info RUST_BACKTRACE=1 cargo run $FEATURES -- --config $CONFIG_PATH &
-SERVER_PID=$!
+PARSEC_PID=$!
 # Sleep time needed to make sure Parsec is ready before launching the tests.
 sleep 5
 
-if [[ $1 = "all" ]]
-then
-	echo "Execute all-providers tests"
-	RUST_BACKTRACE=1 cargo test $FEATURES all_providers
+# Check that Parsec successfully started and is running
+pgrep -f target/debug/parsec >/dev/null
```

Comment:
Good idea! So that if the service does not start successfully we don't even launch the first test.